src/lib.rs

```
@@ -0,0 +1,130 @@
+extern crate diesel;
+extern crate byteorder;
+pub extern crate geo;
+
+use byteorder::{ByteOrder, LittleEndian, BigEndian};
+use std::error::Error;
+use std::io::prelude::*;
+use diesel::pg::{Pg, PgTypeMetadata};
+use diesel::row;
+use diesel::types::{FromSql, ToSql, IsNull, HasSqlType, FromSqlRow};
+use diesel::query_builder::QueryId;
+use std::fmt;
+
+
+#[derive(Debug)]
+struct NotImplemented {}
+impl fmt::Display for NotImplemented {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Not implemented Error")
+    }
+}
+
+impl Error for NotImplemented {
+    fn description(&self) -> &str {
+        "Not implemented"
+    }
+}
+
+
+fn read_u32(bytes: &[u8], big_endian: bool) -> u32 {
+    if big_endian {
+        BigEndian::read_u32(bytes)
+    } else {
+        LittleEndian::read_u32(bytes)
+    }
+}
+
+fn read_f64(bytes: &[u8], big_endian: bool) -> f64 {
+    if big_endian {
+        BigEndian::read_f64(bytes)
+    } else {
+        LittleEndian::read_f64(bytes)
+    }
+}
+
+pub struct Geometry {}
+pub type Geom = geo::Geometry<f64>;
+
+impl FromSql<Geometry, Pg> for Geom {
+    fn from_sql(bytes: Option<&[u8]>) -> Result<Self, Box<Error + Send + Sync>> {
+        let bytes = bytes.unwrap();
```

should you return an error there ?